### PR TITLE
feat: [CDS-83941]: Update functions for Service licenses trend dashboard

### PR DESCRIPTION
**Pull request by mani5h-harness**
## Describe your changes
feat: [CDS-83941]: Update functions for Service licenses trend dashboard

### Changes - 
- updated the query for license trends dashboard on Subscriptions page
- Adde migration for modified functions

Testing notes - 
- Verified query on local data
- Verified migration
- Verified in Pg admin - functions - create script, that functions are updated in DB

**Note - This change is not behind the FF CDS_NG_ACC_ORG_LEVEL_SERVICE_LICENSING_FIX**
Since this change is a migration, I don't see a good way to put this behind a FF.

## Checklist
- [X] I've documented the changes in the PR description.
- [X] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

[CDS-83941]: https://harness.atlassian.net/browse/CDS-83941?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/55837)

### CODE REVIEW

File: 120-ng-manager/src/main/resources/timescale/get_service_instances_by_date_function_timescale.sql [hunk 1]
```diff
@@ -57,7 +57,17 @@ BEGIN
             FROM
                 (SELECT percentile_disc(0.95) WITHIN GROUP (ORDER BY instancesPerServicePerReportedat.instancecount) AS instanceCount,
                         orgid, projectid, serviceid
-                 FROM (SELECT date_trunc('minute', reportedat) AS reportedat, orgid, projectid, serviceid, SUM(instancecount) AS instancecount
+                 FROM (SELECT date_trunc('minute', reportedat) AS reportedat,
```
Comment (RishabhGupta34) [inline 120-ng-manager/src/main/resources/timescale/get_service_instances_by_date_function_timescale.sql:60]: did we test, if this function is getting replaced by the migration?

Comment (mani5h-harness) [reply 120-ng-manager/src/main/resources/timescale/get_service_instances_by_date_function_timescale.sql:60]: Yes, Verified in Pg admin - functions - create script, that functions are updated in DB

**Comment by mani5h-harness** [comment]
trigger uts

File: 120-ng-manager/src/main/resources/timescale/get_service_instances_by_date_function_timescale.sql (hunk 1 above)
Comment (RishabhGupta34) [reply 120-ng-manager/src/main/resources/timescale/get_service_instances_by_date_function_timescale.sql:60]: Using migration as well right? Like old function is present in timescale and then we restart ng manager, the new function gets created

**Comment by mani5h-harness** [comment]
trigger uts

**Review by ivan-83 — APPROVED** [review]

**Comment by mani5h-harness** [comment]
trigger uts

**Comment by mani5h-harness** [comment]
trigger ut5, ut6, ut7, ut9

**Comment by mani5h-harness** [comment]
trigger ut7

**Review by RishabhGupta34 — APPROVED** [review]

**Review by piyush-bhuwalka — APPROVED** [review]

**Comment by mani5h-harness** [comment]
trigger ut7

**Pull request merged**